peer_protocol/peer.go

```diff
@@ -23,59 +59,136 @@ func (p PeerConnection) createMessageBuffer(mt byte, payload []byte) []byte {
 	return buffer
 }
 
-func (p PeerConnection) SendHandshake() {
+func (p *PeerConnection) SendHandshake() {
 	fmt.Println("Handshaking")
-	protocol_type := []byte(Protocol)          // const
+	ProtocolType := []byte(Protocol)           // const
 	reserved := []byte{0, 0, 0, 0, 0, 0, 0, 0} // all zero 8 bytes
-	buf := append(protocol_type, reserved...)
-	buf = append(buf, p.PeerID...)
-	buf = append(buf, p.InfoHash...)
-	l, err := p.Conn.Write(buf)
+	buf := append(ProtocolType, reserved...)
+	buf = append(buf, p.Config.PeerID...)
+	buf = append(buf, p.Config.InfoHash...)
+	_, err := p.Conn.Write(buf)
 	if err != nil {
+		fmt.Printf("Error while handshaking %s", err.Error())
+	} else {
 	}
-	fmt.Println(l)
 }
 
-func (p PeerConnection) IsHandshakeValid(protocol string, hash []byte) bool {
+func (p *PeerConnection) IsHandshakeValid(protocol string, hash []byte) bool {
 	if protocol != protocol {
 		return false
-	} else if bytes.Compare(hash, p.InfoHash) == 0 {
+	} else if bytes.Compare(hash, p.Config.InfoHash) == 0 {
 		return false
 	} else {
 		return true
 	}
 }
 
-func (p PeerConnection) ReceiveHandshake() bool {
+func (p *PeerConnection) ReceiveHandshake() bool {
 	fmt.Println("Handshaking")
-	buf := make([]byte, 68)
-	_, err := p.Conn.Read(buf)
+	buf, err := p.readSocket(68, 10)
```

Comment:
Seems like you have a lot of magic numbers, it might be hard to follow them this way.